external_search.py

```diff
@@ -882,8 +577,748 @@ def create_new_version(cls, search_client, base_index_name, version=None):
             search_client.setup_index(new_index=versioned_index)
             return True
 
+class SearchBase(object):
+
+    @classmethod
+    def _match_range(cls, field, operation, value):
+        """Match a ranged value for a field, using an operation other than
+        equality.
+
+        e.g. _match_range("field.name", "gte", 5) will match
+        any value for field.name greater than 5.
+        """
+        match = {field : {operation: value}}
+        return dict(range=match)
+
+
+class Query(SearchBase):
+    """An attempt to find something in the search index."""
+
+    # When we run a simple query string search, we are matching the
+    # query string against these fields.
+    SIMPLE_QUERY_STRING_FIELDS = [
+        # These fields have been stemmed.
+        'title^4',
+        "series^4",
+        'subtitle^3',
+        'summary^2',
+        "classifications.term^2",
+
+        # These fields only use the standard analyzer and are closer to the
+        # original text.
+        'author^6',
+        'publisher',
+        'imprint'
+    ]
+
+    # When we look for a close match against title, author, or series,
+    # we apply minimal stemming, because we're handling the case where
+    # the user typed something in exactly as is.
+    MINIMAL_STEMMING_QUERY_FIELDS = [
```

Comment:
This is not quite accurate, author has no stemming rather than minimal stemming. I believe minimal stemming only removes plurals, and our minimal analyzer also removes stop words. If you really want exact matches, you should use the standard analyzer rather than minimal.